Src/Particle/AMReX_ParticleMesh.H

```diff
@@ -18,9 +18,12 @@ auto call_f (F const& f,
              GpuArray<Real,AMREX_SPACEDIM> const& plo,
              GpuArray<Real,AMREX_SPACEDIM> const& dxi) noexcept
 {
-    if constexpr (IsCallable<F, decltype(p.m_aos[i]), decltype(fabarr), decltype(plo), decltype(dxi)>::value) {
+    using PTDTypeT = std::remove_const_t<std::remove_reference_t<decltype(p)>>;
```

Comment:
Let's add `#include <type_traits>` to this file for robustness :)

Reply:
isen't ´PTDTypeT´ the same as `PTDType<T_ParticleType, NAR, NAI>`? Also `PTDTypeT::ParticleRefType` and `T_ParticleType`.

Reply:
Right. Maybe it's silly to do it like this, but I just didn't want to introduce typos.